ldk/go/docs/loops.md

```diff
@@ -0,0 +1,51 @@
+# Loops
+
+Loops receive events and use them to generate relevant whispers. Loops choose which events they want to use and which they want to ignore.
+
+## Interface
+
+Writing a loop boils down to writing an implementation for the `Loop` interface.
+
+```go
+type Loop interface {
+	LoopStart(Sidekick) error
+	LoopStop() error
+}
+```
+
+**LoopStart** - The Loop should wait to start operating until this is called. The provided `Sidekick` reference should be stored in memory for continued use.
+
+**LoopStop** - The Loop should stop operating when this is called.
+
+## Subscribing to Sensors
+
+Inside `LoopStart`, you can subscribe to various sensors. Here's an example of subscribing to a couple:
+
+```go
+func (c *Loop) LoopStart(sidekick ldk.Sidekick) error {
+	// ...
+
+	handler := func (text string, err error) {
+		// Respond to sensor event here...
+	}
+
+	if err := sidekick.Clipboard().Listen(l.ctx, handler); err != nil {
+		return err
+	}
+
+	if err := sidekick.UI().ListenSearchbar(l.ctx, handler); err != nil {
+		return err
+	}
+}
+```
+
+Loops do not need to emit whispers in a 1:1 relationship with events. Loops may not use events at all. Loops may only use some events. Loops may keep a history of events and only emit whispers when several conditions are met.
+
+## Lifecycle
+
+1. Sidekick executes plugin process.
```

Comment:
OK - so... there's a bunch of renaming that needs to be done here. I think that'd be covered within SIDE-988 right? 😄 

Soo, could you go through and replace all references to "Sidekick" with "Olive Helps", excluding code references? I'll need to rename the `Sidekick` struct at some point, but anything in docs should be renamed.

Reply:
Sounds good to me - I can rename from "Sidekick" to "Olive Helps".

Just to be clear - I **won't* change the code samples in the docs yet and will wait for that to be done when the references throughout the LDK are changed from `Sidekick` to e.g. `OliveHelps`, right?